// jsonresume-theme-even/dist/index.umd.cjs
(function(d,u){typeof exports=="object"&&typeof module<"u"?u(exports,require("micromark"),require("striptags"),require("feather-icons")):typeof define=="function"&&define.amd?define(["exports","micromark","striptags","feather-icons"],u):(d=typeof globalThis<"u"?globalThis:d||self,u(d.jsonresumeThemeEven={},d.micromark,d.striptags,d.feather))})(this,function(d,u,y,b){"use strict";function t(e,...i){return e.reduce((n,a,r)=>{const o=i[r];return Array.isArray(o)?n+a+o.join(""):o!=null&&!!o!==o?n+a+o:n+a},"")}function l(e,i=!1){const n=u(e);return i?y(n):n}const k=e=>new Date(e).toLocaleDateString("en",{month:"short",year:"numeric",timeZone:"UTC"});function g(e){return t`<time datetime="${e}">${k(e)}</time>`}function w(e=[]){return e.length>0&&t`
    <section id="awards">
      <h3>Awards</h3>
      <div class="stack">
        ${e.map(({awarder:i,date:n,summary:a,title:r})=>t`
          <article>
            <header>
              <h4>${r}</h4>
              <div class="meta">
                ${i&&t`
                  <div>
                    Awarded by <strong>${i}</strong>
                  </div>
                `}
                ${n&&g(n)}
              </div>
            </header>
            ${a&&l(a)}
          </article>
        `)}
      </div>
    </section>
  `}const x=e=>e.replace(/^(https?:|)\/\//,"").replace(/\/$/,"");function h(e,i){return i?e?t`<a href="${e}">${i}</a>`:i:e&&t`<a href="${e}">${x(e)}</a>`}function L(e=[]){return e.length>0&&t`
    <section id="certificates">
      <h3>Certificates</h3>
      <div class="stack">
        ${e.map(({date:i,issuer:n,name:a,url:r})=>t`
          <article>
            <header>
              <h4>${h(r,a)}</h4>
              <div class="meta">
                ${n&&t`
                  <div>
                    Issued by <strong>${n}</strong>
                  </div>
                `}
                ${i&&g(i)}
              </div>
            </header>
          </article>
        `)}
      </div>
    </section>
  `}function p(e,i){return t`${g(e)} – ${i?g(i):"Present"}`}function j(e=[]){return e.length>0&&t`
    <section id="education">
      <h3>Education</h3>
      <div class="stack">
        ${e.map(({area:i,courses:n=[],institution:a,startDate:r,endDate:o,studyType:c,url:m})=>t`
          <article>
            <header>
              <h4>${h(m,a)}</h4>
              <div class="meta">
                ${i&&t`<strong>${i}</strong>`}
                <div>${p(r,o)}</div>
              </div>
            </header>
            ${c&&l(c)}
            ${n.length>0&&t`
              <h5>Courses</h5>
              <ul>
                ${n.map(s=>t`<li>${l(s)}</li>`)}
              </ul>
            `}
          </article>
        `)}
      </div>
    </section>
  `}function f(e,i){return(b.icons[e.toLowerCase()]||b.icons[i.toLowerCase()]).toSvg({width:16,height:16})}const C=e=>Intl.DisplayNames?new Intl.DisplayNames(["en"],{type:"region"}).of(e):e;function z(e={}){const{email:i,image:n,label:a,location:r,name:o,phone:c,profiles:m=[],summary:s,url:$}=e;return t`
    <header class="masthead">
      ${n&&t`<img src="${n}" alt="">`}
      <div>
        ${o&&t`<h1>${o}</h1>`}
        ${a&&t`<h2>${a}</h2>`}
      </div>
      ${s&&t`<article>${l(s)}</article>`}
      <ul class="icon-list">
        ${r?.city&&t`
          <li>
            ${f("map-pin")}
            ${r.city}${r.countryCode&&t`, ${C(r.countryCode)}`}
          </li>
        `}
        ${i&&t`
          <li>
            ${f("mail")}
            <a href="mailto:${i}">${i}</a>
          </li>
        `}
        ${c&&t`
          <li>
            ${f("phone")}
            <a href="tel:${c.replace(/\s/g,"")}">${c}</a>
          </li>
        `}
        ${$&&t`
          <li>
            ${f("link")}
            ${h($)}
          </li>
        `}
        ${m.map(({network:v,url:V,username:W})=>t`
          <li>
            ${f(v,"user")}
            ${h(V,W)}
            ${v&&t`<span class="network">(${v})</span>`}
          </li>
        `)}
      </ul>
    </header>
  `}function I(e=[]){return e.length>0&&t`
    <section id="interests">
      <h3>Interests</h3>
      <div class="grid-list">
        ${e.map(({keywords:i=[],name:n})=>t`
          <div>
            ${n&&t`<h4>${n}</h4>`}
            ${i.length>0&&t`
              <ul class="tag-list">
                ${i.map(a=>t`<li>${a}</li>`)}
              </ul>
            `}
          </div>
        `)}
      </div>
    </section>
  `}function P(e=[]){return e.length>0&&t`
    <section id="languages">
      <h3>Languages</h3>
      <div class="grid-list">
        ${e.map(({fluency:i,language:n})=>t`
          <div>
            ${n&&t`<h4>${n}</h4>`}
            ${i}
          </div>
        `)}
      </div>
    </section>
  `}function q(e={}){const{name:i,summary:n}=e;return t`
    ${i&&t`<title>${i}</title>`}
    ${n&&t`<meta name="description" content="${l(n,!0)}">`}
  `}const R=e=>Intl.ListFormat?new Intl.ListFormat("en").format(e):e.join(", ");function D(e=[]){return e.length>0&&t`
    <section id="projects">
      <h3>Projects</h3>
      <div class="stack">
      ${e.map(({description:i,entity:n,highlights:a=[],name:r,startDate:o,endDate:c,roles:m=[],url:s})=>t`
        <article>
          <header>
            <h4>${h(s,r)}</h4>
            <div class="meta">
              <div>
                ${m.length>0&&t`<strong>${R(m)}</strong>`}
                ${n&&t`at <strong>${n}</strong>`}
              </div>
              <div>${p(o,c)}</div>
            </div>
          </header>
          ${i&&l(i)}
          ${a.length>0&&t`
            <ul>
              ${a.map($=>t`<li>${l($)}</li>`)}
            </ul>
          `}
        </article>
      `)}
      </div>
    </section>
  `}function S(e=[]){return e.length>0&&t`
    <section id="publications">
      <h3>Publications</h3>
      <div class="stack">
        ${e.map(({name:i,publisher:n,releaseDate:a,summary:r,url:o})=>t`
          <article>
            <header>
              <h4>${h(o,i)}</h4>
              <div class="meta">
                ${n&&t`
                  <div>
                    Published by <strong>${n}</strong>
                  </div>
                `}
                ${a&&g(a)}
              </div>
            </header>
            ${r&&l(r)}
          </article>
        `)}
      </div>
    </section>
  `}function T(e=[]){return e.length>0&&t`
    <section id="references">
      <h3>References</h3>
      <div class="stack">
        ${e.map(({name:i,reference:n})=>t`
          <blockquote>
            ${n&&l(n)}
            ${i&&t`
              <p>
                <cite>${i}</cite>
              </p>
            `}
          </blockquote>
        `)}
      </div>
    </section>
  `}function A(e=[]){return e.length>0&&t`
    <section id="skills">
      <h3>Skills</h3>
      <div class="grid-list">
        ${e.map(({keywords:i=[],name:n})=>t`
          <div>
            ${n&&t`<h4>${n}</h4>`}
            ${i.length>0&&t`
              <ul class="tag-list">
                ${i.map(a=>t`<li>${a}</li>`)}
              </ul>
            `}
          </div>
        `)}
      </div>
    </section>
  `}function E(e=[]){return e.length>0&&t`
    <section id="volunteer">
      <h3>Volunteer</h3>
      <div class="stack">
        ${e.map(({highlights:i=[],organization:n,position:a,startDate:r,endDate:o,summary:c,url:m})=>t`
          <article>
            <header>
              <h4>${a}</h4>
              <div class="meta">
                <strong>${h(m,n)}</strong>
                <div>${p(r,o)}</div>
              </div>
            </header>
            ${c&&l(c)}
            ${i.length>0&&t`
              <ul>
                ${i.map(s=>t`<li>${l(s)}</li>`)}
              </ul>
            `}
          </article>
        `)}
      </div>
    </section>
  `}function O(e=[]){return e.length>0&&t`
    <section id="work">
      <h3>Work</h3>
      <div class="stack">
        ${e.map(({description:i,highlights:n=[],location:a,name:r,position:o,startDate:c,endDate:m,summary:s,url:$})=>t`
          <article>
            <header>
              <h4>${o}</h4>
              <div class="meta">
                <div>
                  <strong>${h($,r)}</strong>
                  ${i&&t`<span class="bullet-item">${i}</span>`}
                </div>
                <div>${p(c,m)}</div>
                ${a&&t`<div>${a}</div>`}
              </div>
            </header>
            ${s&&l(s)}
            ${n.length>0&&t`
              <ul>
                ${n.map(v=>t`<li>${l(v)}</li>`)}
              </ul>
            `}
          </article>
        `)}
      </div>
    </section>
  `}function F(e,i){return t`<!DOCTYPE html>
    <html lang="en">
      <head>
        <meta charset="utf-8">
        ${q(e.basics)}
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700&display=swap">
        <style>${i}</style>
      </head>
      <body>
        ${z(e.basics)}
        ${O(e.work)}
        ${E(e.volunteer)}
        ${j(e.education)}
        ${D(e.projects)}
        ${w(e.awards)}
        ${L(e.certificates)}
        ${S(e.publications)}
        ${A(e.skills)}
        ${P(e.languages)}
        ${I(e.interests)}
        ${T(e.references)}
      </body>
    </html>`}const M=`:root{color-scheme:light dark;--color-background: #ffffff;--color-muted: #f3f4f5;--color-primary: #191e23;--color-secondary: #6c7781;--color-accent: #ffb900;--color-link: #46b450;--scale-ratio: 1.25;--scale0: 1rem;--scale1: calc(var(--scale0) * var(--scale-ratio));--scale2: calc(var(--scale1) * var(--scale-ratio));--scale3: calc(var(--scale2) * var(--scale-ratio));--scale4: calc(var(--scale3) * var(--scale-ratio));--scale5: calc(var(--scale4) * var(--scale-ratio))}@media (prefers-color-scheme: dark){:root{--color-background: #191e23;--color-muted: #23282d;--color-primary: #fbfbfc;--color-secondary: #ccd0d4}}*{box-sizing:border-box;margin:0;padding:0}html{font-size:14px}body{background:var(--color-background);color:var(--color-primary);display:grid;font:1em/1.5 Lato,sans-serif;gap:2em;grid-template-columns:[full-start] 1fr [main-start side-start] minmax(min-content,12em) [side-end content-start] minmax(min-content,36em) [main-end content-end] 1fr [full-end];grid-template-rows:auto [content] 0;margin-bottom:4em}body:before{content:"";grid-column:full;grid-row:content}ol,ul{padding-left:1em}:not(.icon-list,.tag-list)>li+li{margin-top:.4em}li::marker,.network{color:var(--color-secondary)}a{color:var(--color-link);text-decoration:none;transition:color .2s}a:focus,a:hover{color:var(--color-accent)}h1,h2,h3,h5{font-weight:400}h1,h2,h3{line-height:1.2}h1{font-size:var(--scale5)}h2{color:var(--color-secondary);font-size:var(--scale4)}h3{color:var(--color-accent);font-size:var(--scale3);grid-column:side;margin-bottom:1rem}h4{font-size:var(--scale2)}h5{font-size:var(--scale1)}h6{font-size:var(--scale0)}blockquote{border-left:.2em solid var(--color-muted);padding-left:1em}cite{color:var(--color-secondary);font-style:inherit}cite:before{content:"\\2014  "}svg{margin-right:.2em;vertical-align:-.2em}.masthead{background:var(--color-muted);display:inherit;gap:inherit;grid-column:full;grid-template-columns:inherit;padding:4em 0;text-align:center}.masthead>*,section{grid-column:main}.masthead>img{border-radius:50%;margin:0 auto;max-width:12em}article>*+*,blockquote>*+*{margin-top:.6em}.meta{color:var(--color-secondary)}.stack{display:grid;gap:1.5em}.icon-list{display:flex;flex-wrap:wrap;gap:.4em 1em;justify-content:center;list-style:none;padding:0}.grid-list{display:grid;gap:1em}.tag-list{display:flex;flex-wrap:wrap;gap:.4em;list-style:none;padding:0}.tag-list>li{background:var(--color-muted);border-radius:.2em;padding:.2em .6em}.bullet-item:not(:first-child):before{content:"\\b7  ";margin:0 .2em}@media print,(min-width: 48em){h3{text-align:right;margin-bottom:inherit}.masthead{text-align:inherit}.masthead>*,section{grid-column:content}.masthead img{grid-column:side;grid-row:span 2;max-width:100%}section{display:contents}.icon-list{flex-direction:column}.grid-list{grid-template-columns:repeat(auto-fit,minmax(calc((100% - 1em)/2),1fr))}}
`,N={mediaType:"print"},U=e=>F(e,M);d.pdfRenderOptions=N,d.render=U,Object.defineProperty(d,Symbol.toStringTag,{value:"Module"})});
